ghost/email-service/lib/EmailRenderer.js

```diff
@@ -278,6 +278,16 @@ class EmailRenderer {
 
                 // Remove the members-only content
                 html = html.slice(0, membersOnlyIndex);
+
+                const $ = cheerio.load(html);
+
+                $('[data-gh-segment]').get().forEach((node) => {
+                    if (node.attribs['data-gh-segment'] !== 'status:free') {
+                        $(node).remove();
+                    }
+                });
+
+                html = $.html();
```

Comment:
Can't we move the part from a little bit down here to above `getTemplateData` instead of repeating it twice? So that it happens for all segments instead of only free members?

\+ maybe add a comment for future reference why we need to remove the segments before we call `getTemplateData`, so that the preheader doesn't contain paid text

Reply:
👍 Updated in https://github.com/TryGhost/Ghost/pull/17137/commits/553fadb1f31277b5f5f92d1318cb9c6663707c4b